scroll_overlay/android/app/src/main/java/io/flutter/scroll_overlay/ScrollOverlayActivity.java

```diff
@@ -3,151 +3,191 @@
+            @Override
+            public void run() {
+                if (flutterEngine == null) {
+                    return;
+                }
+                if (!locked) {
+                    velocitySink.success((double) currentDy * ticksPerSecond);
+                    if (currentDy == 0) {
+                        locked = true;
+                    }
+                    currentDy = 0;
+                }
+                handler.postDelayed(velocityTrackerRunnable, 1000 / ticksPerSecond);
+            }
+        };
 
-    private FlutterView flutterView;
+        void start() {
+            velocityTrackerRunnable.run();
+        }
+
+        void stop() {
+            handler.removeCallbacksAndMessages(null);
+        }
+
+        @Override
+        public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
+            locked = false;
+            currentDy += dy;
+        }
+    }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        FlutterMain.ensureInitializationComplete(getApplicationContext(), null);
         setContentView(R.layout.scroll_overlay_layout);
 
-        flutterView = findViewById(R.id.flutter_view);
-        final FlutterRunArguments args = new FlutterRunArguments();
-        args.bundlePath = FlutterMain.findAppBundlePath();
-        args.entrypoint = "main";
-        flutterView.runFromBundle(args);
+        VelocityTracker velocityTracker = new VelocityTracker();
+
+        // Create flutter view
+        flutterEngine = new FlutterEngine(this);
+        flutterEngine.getDartExecutor().executeDartEntrypoint(
+                DartExecutor.DartEntrypoint.createDefault());
+        FlutterEngineCache
+                .getInstance()
+                .put(engineId, flutterEngine);
+        FlutterView flutterView = findViewById(R.id.flutter_view);
+        flutterView.attachToFlutterEngine(flutterEngine);
+        new EventChannel(flutterEngine.getDartExecutor().getBinaryMessenger(), VELOCITY_CHANNEL).setStreamHandler(
+                new EventChannel.StreamHandler() {
+                    @Override
+                    public void onListen(Object arguments, EventChannel.EventSink events) {
+                        velocitySink = events;
+                        velocityTracker.start();
+                    }
+
+                    @Override
+                    public void onCancel(Object arguments) {
+                        velocityTracker.stop();
+                        velocitySink = null;
+                    }
+                }
+        );
 
-        ListView overlayList = findViewById(R.id.overlay_list);
-        overlayList.setDividerHeight(0);
-        overlayList.setAdapter(new OverlayAdapter());
+        // Create list
+        RecyclerView recyclerView = findViewById(R.id.recycler_view);
+        recyclerView.addItemDecoration(new DividerItemDecoration(this, 0)); // Remove divider
```

Comment:
what part is removing the divider?

Reply:
This line does, I know it's confusing https://stackoverflow.com/a/47119961/9710294